### PR TITLE
Feature/cleanup health update

### PR DESCRIPTION
**Pull request by open-schnick**
Will have a story soon.
Added the remaining keys.

### CODE REVIEW

File: webapp_frontend/src/background/api/api.ts [hunk 1]
```diff
@@ -1,16 +1,25 @@
...
 
 export const userPath: string = '/v1/users';
 
 export const filesystemPath: string = '/v1/filesystem';
 
+enum DataIntegrity {
+    STABLE = "systemHealth/traffic_green.png",
+    POSSIBLE_RISK = "systemHealth/traffic_yellow.png",
+    UNSTABLE = "systemHealth/traffic_red.png"
+}
```
Comment (open-schnick) [inline webapp_frontend/src/background/api/api.ts]: These images still need to be created.

File: webapp_frontend/src/components/pages/Health.tsx [hunk 2]
```diff
@@ -1,14 +1,21 @@
...
+import { logout } from "../../background/api/auth";
+import { getDurationAsString } from "../../background/methods/time";
+import { hasKey } from "../../background/methods/ObjectKeysTS";
 
 export default function Health() {
 
-    const [backendLiveTime, setBackendLiveTime] = useState<number | "not reachable">("not reachable");
-    const [backendUserCount, setBackendUserCount] = useState<number | "not reachable">("not reachable");
+    // TODO: maybe refactor this?
+    const [uptimeInSeconds, setUptimeInSeconds] = useState<number | "not reachable">("not reachable");
```
Comment (open-schnick) [inline webapp_frontend/src/components/pages/Health.tsx]: refactoring this to one object?

File: webapp_frontend/src/components/pages/Health.tsx [hunk 3]
```diff
@@ -49,24 +84,38 @@ export default function Health() {
                 }}
             />
 
-
             <div>
                 <Table striped bordered hover id={"ff-heath-table"}>
```
Comment (open-schnick) [inline webapp_frontend/src/components/pages/Health.tsx]: `heath` ??

**Comment by open-schnick** [comment]
Reduced Renders from **`14` -> `4`**

File: webapp_frontend/src/components/pages/Health.tsx (hunk 2 above)
Comment (open-schnick) [reply webapp_frontend/src/components/pages/Health.tsx]: @me done

**Comment by open-schnick** [comment]
@qvalentin can you fix the tests?

File: webapp_frontend/src/background/api/api.ts [hunk 4]
```diff
@@ -1,19 +1,28 @@
-import Axios, {} from "axios";
-import {constants} from "../constants";
+import Axios, { } from "axios";
+import { constants } from "../constants";
```
Comment (qvalentin) [inline webapp_frontend/src/background/api/api.ts:2]: prtty

File: webapp_frontend/src/background/api/api.ts (hunk 1 above)
Comment (qvalentin) [reply webapp_frontend/src/background/api/api.ts]: @Gimleux

File: webapp_frontend/src/components/pages/Health.tsx [hunk 5]
```diff
@@ -49,24 +69,38 @@ export default function Health() {
...
-                    </tr>
+                        <tr>
+                            <td>Deployment Type</td>
+                            <td>{(systemHealthData === null || systemHealthData.deployment === null) ? errorMsg : systemHealthData.deployment}</td>
+                        </tr>
+                        <tr>
+                            <td>Version</td>
+                            <td>{(systemHealthData === null || systemHealthData.version === null) ? errorMsg : systemHealthData.version}</td>
+                        </tr>
+                        <tr>
```
Comment (qvalentin) [inline webapp_frontend/src/components/pages/Health.tsx]: can we like hide the whole table and show a loading animation, only if the request fails show "not reachable"

**Review by qvalentin — CHANGES_REQUESTED** [review]

File: webapp_frontend/src/components/pages/Health.tsx (hunk 5 above)
Comment (qvalentin) [reply webapp_frontend/src/components/pages/Health.tsx]: https://react-bootstrap.netlify.app/components/spinners/#spinners

**Comment by qvalentin** [comment]
> @qvalentin can you fix the tests?

you changed the table, now the css selector for getting the usercount is not working!

File: webapp_frontend/src/components/pages/Health.tsx (hunk 3 above)
Comment (open-schnick) [reply webapp_frontend/src/components/pages/Health.tsx]: @qvalentin intended?

File: webapp_frontend/src/background/api/api.ts (hunk 1 above)
Comment (qvalentin) [reply webapp_frontend/src/background/api/api.ts]: https://material.io/resources/icons/?icon=traffic&style=baseline

File: webapp_frontend/src/components/pages/Health.tsx (hunk 5 above)
Comment (open-schnick) [reply webapp_frontend/src/components/pages/Health.tsx]: done.

File: webapp_frontend/src/components/basicElements/Loading.tsx [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+import Spinner from "react-bootstrap/Spinner"
+
+
+export function FFLoading(): JSX.Element {
+    return (
+        <Spinner animation="border" role="status">
+            <span className="sr-only">Loading...</span>
+        </Spinner>
+    );
+}
```
Comment (open-schnick) [inline webapp_frontend/src/components/basicElements/Loading.tsx]: Pretty sure this is the wrong place. also this isn't styled properly. (location)

File: webapp_frontend/src/background/api/api.ts [hunk 7]
```diff
@@ -1,38 +1,38 @@
...
-export const filesystemPath: string = '/v1/filesystem';
+export const filesystemPath: string = "/v1/filesystem";
 
 enum DataIntegrity {
-    STABLE = "systemHealth/traffic_green.png",
-    POSSIBLE_RISK = "systemHealth/traffic_yellow.png",
-    UNSTABLE = "systemHealth/traffic_red.png"
+  STABLE = "systemHealth/traffic_green.png",
+  POSSIBLE_RISK = "systemHealth/traffic_yellow.png",
+  UNSTABLE = "systemHealth/traffic_red.png",
```
Comment (open-schnick) [inline webapp_frontend/src/background/api/api.ts]: Did i already mention the fact, that this repo is lacking some config on how to format.

**Comment by open-schnick** [comment]
nice @qvalentin.
TODO:

1. fix the spinner
2. create / get the traffic light images.

**Comment by qvalentin** [comment]
> nice @qvalentin.
> TODO:
> 
> 1. fix the spinner
> 2. create / get the traffic light images.

done

see https://storybook.filefighter.de/ (wait till 14:30)

**Comment by open-schnick** [comment]
Looks very good to me. Thanks @Gimleux and @qvalentin. Its mergable for me.

**Review by qvalentin — APPROVED** [review]

**Pull request merged**